src/TMS/Booking/GUIBindings.php

```diff
@@ -0,0 +1,125 @@
+<?php
+/**
+ * cat-tms-patch start
+ */
+
+use ILIAS\TMS\Wizard;
+
+require_once("Services/Form/classes/class.ilPropertyFormGUI.php");
+require_once("Services/TMS/Booking/classes/class.ilTMSBookingPlayerStateDB.php");
+
+/**
+ * GUI Bindings for TMS-Booking process.
+ *
+ * @author Richard Klees <richard.klees@concepts-and-training.de>
+ */
+class ilTMSBookingGUIBinding extends Wizard\GUIBindings {
+	/**
+	 * @var	ilLanguage
+	 */
+	protected $lng;
+
+	/**
+	 * @var	ilCtrl
+	 */
+	protected $ctrl;
+
+	/**
+	 * @var	mixed
+	 */
+	protected $parent_gui;
+
+	/**
+	 * @var string
+	 */
+	protected $parent_cmd;
+
+	/**
+	 * @var string
+	 */
+	protected $player_title;
+
+	/**
+	 * @var string
+	 */
+	protected $confirm_button_label;
+
+	/**
+	 * @var string
+	 */
+	protected $overview_description;
+
+	final public function __construct(\ilLanguage $lng, \ilCtrl $ctrl, $parent_gui, $parent_cmd, $player_title, $confirm_button_label, $overview_description) {
+		assert('is_object($parent_gui)');
+		assert('is_object($parent_cmd)');
+		assert('is_string($confirm_button_label)');
+		assert('is_string($player_title)');
+		assert('is_string($overview_description)');
+		$this->lng = $lng;
+		$this->ctrl = $ctrl;
+		$this->lng->loadLanguageModule('tms');
+		$this->parent_gui = $parent_gui;
+		$this->parent_cmd = $parent_cmd;
+		$this->player_title = $player_title;
+		$this->confirm_button_label = $confirm_button_label;
+		$this->overview_description = $overview_description;
+	}
+
+	/**
+	 * @inheritdocs
+	 */
+	protected function getForm() {
+		$form = new ilPropertyFormGUI();
+		$form->setFormAction($this->g_ctrl->getFormAction($this));
+		$form->setShowTopButtons(true);
+		return $form;
+	}
+
+	/**
+	 * @inheritdocs
+	 */
+	protected function txt($id) {
```

Comment:
Ich mag das hier irgendwie immer noch nicht. Klar sparen wir dadurch an public functionen, aber wenn wir jetzt schon eine eigene GUI dafür haben, dann sollte man mal Übergelgen ob "gui_binding->getPlayerTitle()" nicht für die Zukunft besser zu lesen ist, als "gui_binding->txt("title")" wo wir dann wieder gucken müssen zu was "title" umgemünzt wird.

Außerdem muss "aborted" auch noch umgearbeitet werden, da beim Abbrechen von Wartelistenbuchung oder FK-Buchung eine andere Meldung erscheinen muss. Dann bleiben wieder nur die Buttonlabel.

Reply:
Ich verstehe deinen Punkt. Auf der anderen Seite: Wollen wir für jeden Wunsch analog zu "hier Titel anders" eine neue Methode einführen? Das scheint mir eher unlesbar. Gibt es vllt. einen Zwischenweg, z.B. Konstanten für die verwendeten Sprach-IDs einführen? Das ist ja im Grunde ein allgemeines Problem, was wir immer wieder bekommen werden, wenn wir solche Arten von Klassen ("generalisierte GUI-Klassen") haben.

Reply:
Das Problem ist ja weniger das Ding wo die Ausgabe nun her kommt. Es is dieser blöde Switch zwischen den GUI's. Mit den Bindings ist er ja nun nur an eine Stelle gewandert.

Dadurch das die Ausgabe nun in den Bindings sind, und hier die Werte ja eh gesetzt werden, finde ich getter dafür nicht so unübersichtlich, bzw. später schneller wieder zu finden.

Wie bekommen wir denn die Konstanten aus den Playern in die Bindings? Das bräuchten wir ja für jeden Fall eigene Bindings. Also WaitingBinding, SelfBookBinding, SuperiorBookBinding....